packages/catalog-process/src/services/catalogService.ts

```diff
@@ -43,14 +48,109 @@ const assertRequesterAllowed = (
   }
 };
 
+const assertDescriptorIsDraft = (descriptor: Descriptor): Descriptor => {
+  if (descriptor.state !== "DRAFT") {
+    throw notValidDescriptor(descriptor.id, descriptor.state.toString());
+  }
+  return descriptor;
+};
+
 const retrieveEService = async (
   eServiceId: string
 ): Promise<WithMetadata<CatalogItem>> => {
-  const eservice = await readModelGateway.getCatalogItemById(eServiceId);
-  if (eservice === undefined) {
+  const eService = await readModelGateway.getCatalogItemById(eServiceId);
+  if (eService === undefined) {
     throw eServiceNotFound(eServiceId);
   }
-  return eservice;
+  return eService;
+};
+
+const retrieveDescriptor = async (
+  descriptorId: string,
+  eService: WithMetadata<CatalogItem>
+): Promise<Descriptor> => {
+  const descriptor = eService.data.descriptors.find(
+    (d: Descriptor) => d.id === descriptorId && d.state === "DRAFT"
+  );
+
+  if (descriptor === undefined) {
+    throw new CatalogProcessError(
+      `Descriptor with id ${descriptorId} of EService ${eService.data.id} not found`,
+      ErrorTypes.EServiceDescriptorNotFound
+    );
+  }
+
+  return descriptor;
+};
+
+const updateDescriptorState = (
+  descriptor: Descriptor,
+  updateESErviceDescriptorState: EServiceDescriptorState
+): Descriptor => {
+  const descriptorStateChanges = [
+    descriptor.state,
+    updateESErviceDescriptorState.toString(),
```

Comment:
is `toString()` needed?

Reply:
I already refactor this in my PR https://github.com/pagopa/interop-be-monorepo/pull/28/files#diff-77a5bad1f3b877e7bfc968a9b423c16649fca358d54286d743fe34fa555564ceL92